src/coreclr/vm/memberload.cpp

```diff
@@ -367,16 +367,24 @@ void MemberLoader::GetDescFromMemberRef(ModuleBase * pModule,
         {
             //
             // <NICE> this is duplicated logic GetFieldDescByIndex </NICE>
-            //
-            INDEBUG(mdFieldDef token = pFD->GetMemberDef();)
+            //          
 
-            DWORD pos = static_cast<DWORD>(pFD - (pMT->GetApproxFieldDescListRaw() + pMT->GetNumIntroducedInstanceFields()));
-            _ASSERTE(pos >= 0 && pos < pMT->GetNumStaticFields());
+            INDEBUG(mdFieldDef token = pFD->GetMemberDef();)
 
-            pFD = pMT->GetGenericsStaticFieldDescs() + pos;
-            _ASSERTE(pFD->GetMemberDef() == token);
-            _ASSERTE(!pFD->IsSharedByGenericInstantiations());
-            _ASSERTE(pFD->GetEnclosingMethodTable() == pMT);
+            DWORD pos = static_cast<DWORD>(pFD - (pMT->GetApproxFieldDescListRaw() + pMT->GetNumStaticFields()));
+            if(!(pos >= 0 && pos < pMT->GetNumStaticFields()))
+            {
+                DWORD fdIndex = pMT->GetIndexForFieldDesc(pFD);
+                PTR_FieldDesc pFD = pMT->GetFieldDescByIndex(fdIndex);
```

Comment:
Are PTR_FieldDesc types only used within the DAC?

Reply:
They are used whenever the code in question is being compiled as both DAC and non-DAC code. If it is only one or the other, then it is typically not needed. The type is used to being a remote pointer when building within the DAC or a local pointer in a non-DAC compile. The remote pointer transparently reads the address in the debuggee process (non-DAC) as opposed to the process loading the DAC which is the debugger.

In this case, we are removing the redeclaration because it is shadowing the `FieldDesc* pFD = ...` on line 361. This code needs to update that local not a new one.